pkg/minishift/util/set_env_driver_options.go

```diff
@@ -0,0 +1,30 @@
+package util
+
+import (
+	"github.com/docker/machine/libmachine/drivers"
+	"os"
+	"strings"
+)
+
+func SetEnvDriverOptions(d drivers.Driver) error {
+	supportedFlags := d.GetCreateFlags()
+
+	checkFlags := &drivers.CheckDriverOptions{
+		FlagsValues: map[string]interface{}{},
+		CreateFlags: supportedFlags,
+	}
+
+	for _, flag := range supportedFlags {
+		flagName := flag.String()
+		flagEnv := strings.Replace(strings.ToUpper(flagName), "-", "_", -1)
```

Comment:
-1, why do you  do any conversion here. This might or might not work. There is no enforced standard on how the env variable looks like. _GetCreateFlags_ returns a slice of _[]mcnflag.Flag_. Each instance of mcnflag.Flag has a public member _EnvVar_. So you don't need to convert anything. You literally iterate over the slice, check whether the flag has a value for _EnvVar_ (it might not have, see KVM driver) and use this flag name to call os.Getenv.

Otherwise this is the right approach.